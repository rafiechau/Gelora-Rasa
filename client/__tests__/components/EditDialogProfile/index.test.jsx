import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import EditProfileDialog from '@components/EditDialogProfile';

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  useNavigate: () => jest.fn(),
}));

jest.mock('react-intl', () => ({
  FormattedMessage: jest.fn(({ id }) => <span>{id}</span>),
  injectIntl: (Component) => (props) => <Component {...props} intl={{ formatMessage: (msg) => msg.id }} />,
}));

describe('EditProfileDialog', () => {
  const mockOnClose = jest.fn();
  const mockProfileUser = { firstName: 'John', lastName: 'Doe' };

  beforeEach(() => {
    render(
      <EditProfileDialog
        open
        onClose={mockOnClose}
        intl={{ formatMessage: (msg) => msg.id }}
        profileUser={mockProfileUser}
      />
    );
  });

  test('Should render correctly', () => {
    expect(screen.getByTestId('edit-profile-dialog')).toBeInTheDocument();
  });

  test('Should call onClose when close button is clicked', () => {
    fireEvent.click(screen.getByTestId('close-button'));
    expect(mockOnClose).toHaveBeenCalled();
  });

  test('Should call onSubmit when submit button is clicked', () => {
    fireEvent.click(screen.getByTestId('submit-button'));
  });

  test('Should match snapshot', () => {
    const dialog = screen.getByTestId('edit-profile-dialog');
    expect(dialog).toMatchSnapshot();
  });
});
