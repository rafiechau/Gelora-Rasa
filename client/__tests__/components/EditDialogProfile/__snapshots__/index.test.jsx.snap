// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfileDialog Should match snapshot 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="edit-profile-dialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":r3:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-tlc64q-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        data-testid="dialog-title"
        id=":r3:"
      >
        <span>
          app_header_update_profile
        </span>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-testid="close-button"
          style="position: absolute; right: 8px; top: 8px;"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseSharpIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        data-testid="dialog-content"
      >
        <form
          class="MuiBox-root css-h4tqr2"
          novalidate=""
        >
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="labelInput"
              for="firstName"
            >
              app_user_firstName
              <input
                class="input undefined"
                id="firstName"
                name="firstName"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="labelInput"
              for="lastName"
            >
              app_user_lastName
              <input
                class="input undefined"
                id="lastName"
                name="lastName"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="labelInput"
              for="new_password"
            >
              app_user_password (Optional)
              <input
                class="input undefined"
                id="new_password"
                name="new_password"
                type="password"
                value=""
              />
              <label
                class="showPassword"
                for="show"
              >
                <input
                  id="show"
                  name="show"
                  type="checkbox"
                />
                <span>
                  app_user_password_show
                </span>
              </label>
            </label>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
            data-testid="submit-button"
            tabindex="0"
            type="submit"
          >
            <span>
              app_btn_update_profile
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;
